core/primitives/src/crypto/signer.rs

```diff
@@ -198,36 +201,53 @@ impl InMemorySigner {
         let mut rng = OsRng::new().expect("Unable to generate random numbers");
         let account_id: String =
             rng.sample_iter(&Alphanumeric).filter(|c| c.is_ascii_alphabetic()).take(10).collect();
-        let (public_key, secret_key) = signature::get_key_pair();
+        let (public_key, secret_key) = get_key_pair();
         let bls_secret_key = BlsSecretKey::generate();
         let bls_public_key = bls_secret_key.get_public_key();
         Self { account_id, public_key, secret_key, bls_public_key, bls_secret_key }
     }
 }
 
+use cached::{cached, SizedCache};
+
+/// We store signing in the LRU cache.
+cached!{
+  TRANSACTION_CACHE: SizedCache<(Vec<u8>, SecretKey), Signature> = SizedCache::with_size(1_000_000);
```

Comment:
it's not ideal to store the secret key everywhere, especially a million copies
can this cache be a member of Signer?

Reply:
Fixed. Also added caching for BLS aggregation.